src/fn/edit-contacts.js

```diff
@@ -0,0 +1,169 @@
+const minimist = require('minimist');
+const environment = require('../lib/environment');
+const fs = require('../lib/sync-fs');
+const path = require('path');
+const { warn, info } = require('../lib/log');
+const pouch = require('../lib/db');
+const toDocs = require('./csv-to-docs');
+
+module.exports = ()=> {
+  const args = parseExtraArgs(environment.pathToProject, environment.extraArgs);
+  const db = pouch();
+  const docDirectoryPath = args.docDirectoryPath;
+  fs.mkdir(docDirectoryPath);
+  const saveJsonDoc = doc => fs.write(`${docDirectoryPath}/${doc._id}.doc.json`, toDocs.toSafeJson(doc) + '\n');
+
+  const csvDir = `${environment.pathToProject}/csv`;
+  if(!fs.exists(csvDir)) {
+    warn(`No csv directory found at ${csvDir}.`);
+    return Promise.resolve();
+  }
+
+  return fs.recurseFiles(csvDir)
+  .filter(name => name.endsWith('.csv'))
+  .reduce((promiseChain, csv) =>
+    promiseChain
+    .then(() => {
+      info('Processing CSV file:', csv, '…');
+
+
+      const nameParts = fs.path.basename(csv).split('.');
+      const prefix = nameParts[0];
+      info('Processing CSV file:', prefix, '…');
+
+      switch(prefix) {
+        case 'contact':  return downloadDocs(csv, getIDs(csv),db, args);
+        default: throw new Error(`Unrecognised CSV type ${prefix} for file ${csv}`);
+      }
+    })
+    .then(docs => addToModel(csv, docs)),
+    Promise.resolve())
+
+  .then(() => model.exclusions.forEach(toDocs.removeExcludedField))
+  .then(() => Promise.all(Object.values(model.docs).map(saveJsonDoc)));
+};
+
+const model = {
+  csvFiles: {},
+  docs: {},
+  exclusions: [],
+  users: []
+};
+
+const addToModel = (csvFile, docs) => {
+    csvFile = csvFile.match(/^(?:.*[\/\\])?csv[\/\\](.*)\.csv$/)[1]; // eslint-disable-line no-useless-escape
+    model.csvFiles[csvFile] = docs;
+    docs.forEach(doc => {
+      model.docs[doc._id] = doc;
+    });
+  };
+
+function getIDs(csv) {
+    const { rows, cols } = fs.readCsv(csv);
+    var index = cols.indexOf('uuid');
+    if (index === -1){
+     throw Error('missing "uuid" column.');
+   }
+   return rows.map((item) => item[index]);
+}
+
+function processContacts(contactType, csv, ids, contactDocs, args){
```

Comment:
I don't think you need to `contactType` parameter here. It's hardcoded to `contact` and then never used in the next function that it's passed to.

Reply:
Makes sense. This has now been removed.